lib/python/ray/services.py

```diff
@@ -72,6 +72,17 @@ def cleanup():
 def all_processes_alive():
   return all([p.poll() is None for p in all_processes])
 
+def get_head_node_ip_address():
+  s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+  s.connect(("8.8.8.8", 53))
+  return s.getsockname()[0]
+
+def get_node_ip_address(redis_address):
```

Comment:
this doesn't have to be named redis_address, how about a more general name, e.g., host_service_addr, host_service_port . The advantage is that the caller will not be led to believe that they need redis_addr to get the host IP.